spec/unit/ssl/state_machine_spec.rb

```diff
@@ -63,6 +63,25 @@
     end
   end
 
+  context 'when locking' do
+    it "should lock prior to running the state machine" do
+      allow_any_instance_of(Puppet::X509::CertProvider).to receive(:load_cacerts).and_return(cacerts)
```

Comment:
The test is labeled 'prior to' - should we check the order of lock vs loading?

Reply:
yeah that'd be better, let me update that.

Reply:
updated